VsoBuildStepSources/SonarQubePostBuild/SonarQubePostBuild.ps1

```diff
@@ -13,4 +13,27 @@ if (!$sonarMsBuildRunnerPath -or ![System.IO.File]::Exists($sonarMsBuildRunnerPa
 Write-Verbose -Verbose "Executing $sonarMsBuildRunnerPath without arguments"
 Invoke-BatchScript $sonarMsBuildRunnerPath 
 
+$agentBuildDirectory = Get-Variable $distributedTaskContext "Agent.BuildDirectory"
+
+if (!$agentBuildDirectory)
+{
+    throw "Could not retrieve the Agent.BuildDirectory variable";
```

Comment:
Now that I think about it this is very bad as user message. I'll change it to "Internal error - could not determine the build directory path on the agent"